elements/lisk-transactions/src/base_transaction.ts

```diff
@@ -400,87 +361,51 @@ export abstract class BaseTransaction {
 			sortKeysAscending(keys.optionalKeys);
 			// Sign with all keys
 			for (const aKey of [...keys.mandatoryKeys, ...keys.optionalKeys]) {
+				const publicKeyStr = bufferToHex(aKey as Buffer);
 				// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-				if (keysAndPassphrases[aKey]) {
-					const { passphrase } = keysAndPassphrases[aKey];
+				if (keysAndPassphrases[publicKeyStr]) {
+					const { passphrase } = keysAndPassphrases[publicKeyStr];
 					this.signatures.push(
 						signData(transactionWithNetworkIdentifierBytes, passphrase),
 					);
 				} else {
 					// Push an empty signature if a passphrase is missing
-					this.signatures.push('');
+					this.signatures.push(Buffer.from(''));
 				}
 			}
-			this._id = getId(this.getBytes());
```

Comment:
Since sign will change the id, probably it's good to have this one, but u can do simply
```
this._id = hash(this.getBytes());	
```